broadlink/switch.py

```diff
@@ -216,3 +216,97 @@ def get_energy(self) -> int:
         check_error(response[0x22:0x24])
         payload = self.decrypt(response[0x38:])
         return int(hex(payload[0x07] * 256 + payload[0x06])[2:]) + int(hex(payload[0x05])[2:]) / 100.0
+
+
+class sp4(device):
+    """Controls a Broadlink SP4."""
+
+    def __init__(self, *args, **kwargs):
+        """Initialize the controller."""
+        device.__init__(self, *args, **kwargs)
+        self.type = "SP4"
+
+    def set_power(self, state):
+        """Set the power state of the device"""
+        return self.set_state(pwr=int(bool(state)))
+
+    def set_nightlight(self, state):
+        """Set the night light state of the device"""
+        return self.set_state(ntlight=int(bool(state)))
+
+    def set_state(
+        self,
+        pwr=None,
+        ntlight=None,
+        indicator=None,
+        ntlbrightness=None,
+        maxworktime=None,
+    ):
+        """Set state of device"""
+        data = {}
+        if pwr is not None:
+            data["pwr"] = int(bool(pwr))
```

Comment:
Remove unnecessary conversions. The client should provide an input according to the annotations.